### DISCUSSION TITLE
future warning

### DISCUSSION THREAD

**Discussion by qn222ad** [highfestiva/finplot#259] [Q&A]
I found the running had this issue:
__init__.py:2432: FutureWarning: casting datetime64[ns] values to int64 with .astype(...) is deprecated and will raise in a future version. Use .view(...) instead.
  return t.astype('int64')
The root cause is from this function:

def _pdtime2epoch(t):
    if isinstance(t, pd.Series):
        if isinstance(t.iloc[0], pd.Timestamp):
            return t.astype('int64')
        h = np.nanmax(t.values)
        if h < 1e10:  # handle s epochs
            return (t * 1e9).astype('int64')
        if h < 1e13:  # handle ns epochs
            return (t * 1e6).astype('int64')
        if h < 1e16:  # handle us epochs
            return (t * 1e3).astype('int64')
        return t.astype('int64')
    return t

Can I ask how to fix it?
Thanks

**Comment by highfestiva** [comment]
Thanks for the headsup, fixed in commit bdd5599.
